### PR TITLE
Update docker images to use upstream's colcon yaml

### PR DESCRIPTION
**Pull request by diegoferigo**
After #237, we no longer need to use a custom branch and we can depend directly from upstream. This PR updates the Dockerfile used to generate the images for our CI/CD pipeline to download the official colcon yaml for Citadel.

I also disabled the static compilation of ign-gazebo since it does not work properly as discussed in https://github.com/dic-iit/gazebo-scenario-plugins/pull/2#issuecomment-684539294.

### CODE REVIEW

**Comment by traversaro** [comment]
Eventually this will be replaced by assuming that the correct binaries are already installed in the system via the `apt` `.deb` packages provided by the OpenRobotics repo, right?

**Review by traversaro — APPROVED** [review]

**Comment by diegoferigo** [comment]
In general our policy is to stay compatible with upstream's binaries with the `master` branch, and currently those CI/CD pipelines already use the released packages (then, master is out of sync waiting the v1 release, but this is another story). Instead, `devel` should only be compatible with any upstream branch, not necessarily released / tagged versions. This helps us to keep the pace with upstream and get either the most recent features of the release used in master (Citadel right now) or test / prepare the update to new releases (like Dome).

In short, `devel` will always use a colcon workspace and it will possibly get broken if there are breaking changes. Within the same upstream release, upstream guidelines about API and ABI stability are already quite robust and should prevent that, but never say never :)
